src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionContext.cs

```diff
@@ -95,7 +97,24 @@ public HttpConnectionContext(string id, IDuplexPipe transport, IDuplexPipe appli
 
         public ClaimsPrincipal User { get; set; }
 
-        public override IDictionary<object, object> Items { get; set; } = new ConnectionItems(new ConcurrentDictionary<object, object>());
+        public override IDictionary<object, object> Items
+        {
+            get
+            {
+                if (_items == null)
+                {
+                    lock (_itemsLock)
+                    {
+                        if (_items == null)
+                        {
+                            _items = new ConnectionItems(new ConcurrentDictionary<object, object>());
+                        }
+                    }
+                }
+                return _items;
+            }
+            set => _items = value ?? throw new ArgumentNullException(nameof(value));
```

Comment:
Should `Items` be settable? If it threw an error then the double check lock could probably be replaced with a `Lazy<ConnectionItems>`.

Reply:
No lazy plz. This is fine.

Reply:
le approve?